infra/modules/gcp-host/main.tf

```diff
@@ -175,7 +177,8 @@ module "api_connector" {
 
   secret_bindings = merge(
     local.secrets_bound_as_env_vars[each.key],
-    module.psoxy.secrets
+    module.psoxy.secrets,
+    module.psoxy.artifact_repository
```

Comment:
Need some dependency here. `artifactrepository` API needs to be activated before running this, so:
1) the output variable has a dependency of that
2) we *force* the dependency here in the module too.

Otherwise following error can happen:

`Error while updating cloudfunction configuration: googleapi: Error 403: Cloud Functions uses Artifact Registry to store function docker images. Artifact Registry API is not enabled in your project. To enable the API, visit https://console.developers.google.com/apis/api/artifactregistry.googleapis.com/overview?project=[REDACTED] or use the gcloud command 'gcloud services enable artifactregistry.googleapis.com' then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry., forbidden
`

And it will require two deployments for doing the same: one for activating the API and then, a second one for running it without errors (to avoid the delay between the terraform service and the real activation in the API)